tests/basic/pure_test.go

```diff
@@ -5130,3 +5130,88 @@ var _ = Describe("{ValidatePodNameinVolume}", func() {
 		AfterEachTest(contexts)
 	})
 })
+var _ = Describe("{RestartPxandRestartNode}", func() {
+	/*
+	   https://purestorage.atlassian.net/browse/PTX-24483
+	   1.Deploy Applications
+	   2.Validate Applications are Deployed
+	   3.Restart Portworx service on few nodes
+	   4.once portworx is up on the node,immediately reboot the node
+	   5.Make sure Both portworx and node are up.
+	   6.Validate the Applications are running
+	*/
+	var contexts []*scheduler.Context
+	JustBeforeEach(func() {
+		StartTorpedoTest("RestartPxandRestartNode",
+			"Restart Portworx and Restart Node", nil, 0)
+	})
+	itLog := "RestartPxandRestartNode"
+	It(itLog, func() {
+		log.InfoD(itLog)
+		stNodes := node.GetStorageNodes()
+		selectedNodesForReboot := stNodes[:len(stNodes)/2]
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			taskName := "restartpxandrebootnode"
+			Provisioner := fmt.Sprintf("%v", portworx.PortworxCsi)
+			context, err := Inst().S.Schedule(taskName, scheduler.ScheduleOptions{
+				AppKeys:            Inst().AppList,
+				StorageProvisioner: Provisioner,
+				Namespace:          taskName,
+			})
+			log.FailOnError(err, "Failed to schedule application of %v namespace", taskName)
+			contexts = append(contexts, context...)
+		}
+		ValidateApplications(contexts)
+		defer DestroyApps(contexts, nil)
+		stepLog := "Restart Portworx Service on few nodes and once portworx is up, immediately reboot the node"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			for _, nodeToReboot := range selectedNodesForReboot {
+				log.InfoD("Stopping Portworx Service on Node [%v]", nodeToReboot.Name)
+				err := Inst().V.StopDriver([]node.Node{nodeToReboot}, false, nil)
+				log.FailOnError(err, "Failed to stop portworx on node [%v]", nodeToReboot.Name)
+				err = Inst().V.WaitDriverDownOnNode(nodeToReboot)
```

Comment:
change the logic to label the nodes with px/service=restart and then reboot the same node